msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ClientCertificate.java

```diff
@@ -119,9 +126,15 @@ static ClientCertificate create(final PrivateKey key, final X509Certificate publ
         return new ClientCertificate(key, Arrays.asList(publicKeyCertificate));
     }
 
-    private static byte[] getHash(final byte[] inputBytes) throws NoSuchAlgorithmException {
+    private static byte[] getHashSha1(final byte[] inputBytes) throws NoSuchAlgorithmException {
         final MessageDigest md = MessageDigest.getInstance("SHA-1");
         md.update(inputBytes);
         return md.digest();
     }
+
+    private static byte[] getHash(final byte[] inputBytes) throws NoSuchAlgorithmException {
```

Comment:
nit: consider naming it sha2 or naming the sha1 version as "deprecated" or "legacy"

Reply:
In the latest commit I simply have "Sha256" or "Sha1" in the method names, and added comments explaining why SHA-1 is still used in certain flows.